src/main/kotlin/com/sourcegraph/cody/chat/ChatSession.kt

```diff
@@ -155,8 +157,28 @@ private constructor(private val project: Project, newSessionId: CompletableFutur
               extensionMessage.isMessageInProgress == false) {
             getCancellationToken().dispose()
           } else {
-            if (lastMessage?.text != null && extensionMessage.chatID != null) {
-              addAssistantResponseToChat(lastMessage.text, lastMessage.displayText)
+
+            val penultimateMessage = extensionMessage.messages?.dropLast(1)?.lastOrNull()
+            if (extensionMessage.chatID != null) {
+              if (penultimateMessage != null) {
+                val messageId =
+                    UUID.nameUUIDFromBytes(
+                        (extensionMessage.messages.count() - 1).toString().toByteArray())
```

Comment:
Why `-1`? It is different formula than used in `addAssistantResponseToChat`, this will cause us issues.